### DISCUSSION TITLE
Dedicated server authentication failed when using reverse proxy.

### DISCUSSION THREAD

**Discussion by VelonacepsCalyxEggs** [FakeFishGames/Barotrauma#14417] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Hello There! I was trying to make some of my friends join my server (due to regional restrictions, they can't join the server the normal way), so I've decided to forward traffic to my server through a VPS, always did that with other games, and it worked flawlessly. With Barotrauma however... when a player tries to join through the VPS' ip address, it gives the "Connection authentication failed", I've tried setting the  RequireAuthentication parameter in serversettings.xml to false, but it didn't do anything.  Any ideas on how to solve this problem? I am desperate, I've been bashing my head on this problem for maybe... uhh... 3~ ish hours, nothing helped...

I also tried using external reverse proxy services, they give the same error as described above.

### Reproduction steps

1. Start a dedicated server.
2. Try to connect through a proxy (like playit.gg, doesn't take much time to set it up)
3. Recieve the "Connection authentication failed" error.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted using a dedicated server

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
This bug may also be related to an issue that is currently marked as closed:
https://github.com/FakeFishGames/Barotrauma/issues/13049
```

**Comment by rpoxo** [comment]
Same happening for me, have VPS with public address and wireguard tunnel from game server(baroserver.wg1, resolves to 10.0.1.2 on vps) behind NATs, and tried proxy traffic through nginx.
```
stream {
    server {
            listen 27015 udp;
            proxy_pass baroserver.wg1:27015;
    }

    server {
            listen 27016 udp;
            proxy_pass baroserver.wg1:27016;
    }
}
```
Now, if i connect another client to wireguard network so that i can ping server(10.0.1.2) - then direct connection to game server wireguard's ip works and game connects fine.
However attempting to connect via VPS public ip(which running nginx as reverse proxy) gives "authentication failed" error.
Running Windows dedicated server(not steamcmd), 1.5.9.1 (ReleaseWindows, branch release/summer-update-hotfix-2, revision bedd89e25a), RequireAuthentication="False"

Issue may not existed in 1.0.13.2(i've attempted to do the same back in time)

**Reply by VelonacepsCalyxEggs** [reply]
I am glad I am not the only one having this issue! Hopefully this is going to get resolved soon!

**Comment by rpoxo** [comment]
The workaround i've found is to have game port forwarded through iptables + masquerade server traffic(`AllowedIPs = 0.0.0.0/0`) via said vps with public ip.

`enp38s0` is vps interface with public ip, `wg2` is tunnel, `10.0.2.5` is a internal game server address, wireguard conf as follows:
```
PostUp = ufw route allow in on wg2 out on enp38s0
PostUp = wg set %i private-key /etc/wireguard/wg2.key; iptables -A FORWARD -i wg2 -j ACCEPT; iptables -t nat -A POSTROUTING -o enp38s0 -j MASQUERADE
PostUp = iptables -t nat -A PREROUTING -i enp38s0 -p udp --dport 27015 -j DNAT --to-destination 10.0.2.5:27015
PostUp = iptables -t nat -A PREROUTING -i enp38s0 -p udp --dport 27016 -j DNAT --to-destination 10.0.2.5:27016

PreDown = ufw route delete allow in on wg2 out on enp38s0
PostDown = iptables -D FORWARD -i wg2 -j ACCEPT; iptables -t nat -D POSTROUTING -o enp38s0 -j MASQUERADE
PostDown = iptables -t nat -D PREROUTING -i enp38s0 -p udp --dport 27015 -j DNAT --to-destination 10.0.2.5:27015
PostDown = iptables -t nat -D PREROUTING -i enp38s0 -p udp --dport 27016 -j DNAT --to-destination 10.0.2.5:27016
```

Not applicable to every case unfortunately.

**Comment by VelonacepsCalyxEggs** [comment]
Also may be related to this:
https://github.com/FakeFishGames/Barotrauma/discussions/14343
Maybe the RequireAuthentication parameter just doesn't work?

**Comment by 3RDNature** [comment]
Still having this problem now (1.6.19.1), I'm sure RequireAuthentication is False, still cant' access server through reverse proxy, even didn`t work on NAT VPS (I'm sure portforward is working, and server can get the raw IP from client).
